tests/unit/core/testErrorHandler.js

```diff
@@ -357,4 +393,68 @@ describe('errorHandler', () => {
       expect(wrapper.find(ErrorList)).toHaveLength(1);
     });
   });
+
+  describe('withFixedErrorHandler', () => {
+    const createFixedErrorComponent = (params = {}) => {
+      return createWrappedComponent({
+        decorator: withFixedErrorHandler,
+        fileName: '/path/to/src/SomeComponent/index.js',
+        ...params,
+      });
+    };
+
+    it('throws an error when `extractId` is missing', () => {
+      expect(() => {
+        createFixedErrorComponent({
+          extractId: null,
+        });
+      }).toThrow('`extractId` is required and must be a function.');
+    });
+
+    it('throws an error when `fileName` is not supplied', () => {
+      expect(() => {
+        createFixedErrorComponent({
+          fileName: undefined,
+          extractId: {},
+        });
+      }).toThrow('`fileName` parameter is required.');
+    });
+
+    it('throws an error when `extractId` is not a function', () => {
+      expect(() => {
+        createFixedErrorComponent({
+          extractId: {},
+        });
+      }).toThrow('`extractId` is required and must be a function.');
+    });
+
+    it('creates an error handler with a fixed ID', () => {
+      const { component } = createFixedErrorComponent({
+        extractId: () => 'unique-id-based-on-props',
+      });
+      const errorHandler = component.props.errorHandler;
+      expect(errorHandler).toBeInstanceOf(ErrorHandler);
+      expect(errorHandler.id)
+        .toEqual('src/SomeComponent/index.js-unique-id-based-on-props');
```

Comment:
Unlike the other tests, this one relies on `fileName` in its assertion. Thus, you should declare it explicitly when calling `createFixedErrorComponent()`. Relying on implicit defaults can make debugging failed assertions difficult.